splice_spark2/src/main/scala/com/splicemachine/nsds/kafka/KafkaTopics.scala

```diff
@@ -17,16 +17,47 @@ package com.splicemachine.nsds.kafka
 
 import com.splicemachine.primitives.Bytes
 import java.security.SecureRandom
+import java.util.concurrent.LinkedTransferQueue
+import java.util.concurrent.atomic.AtomicBoolean
+
+import scala.collection.JavaConverters._
+
 import edu.umd.cs.findbugs.annotations.SuppressFBWarnings
 
 @SerialVersionUID(20200518241L)
 @SuppressFBWarnings(value = Array("NP_ALWAYS_NULL"), justification = "Field 'unused' initialization is not null")
-class KafkaTopics(kafkaServers: String, defaultNumPartitions: Int = 1, defaultRepFactor: Short = 1) extends Serializable {
-  val admin = new KafkaAdmin(kafkaServers)
-  val unneeded = collection.mutable.HashSet[String]()
+class KafkaTopics(
+    kafkaServers: String, 
+    defaultNumPartitions: Int = 1, 
+    defaultRepFactor: Short = 1,
+    continuousCleanup: Boolean = true
+  ) extends Serializable
+{
+  private val admin = new KafkaAdmin(kafkaServers)
+
+  private val unneeded = new LinkedTransferQueue[String]()
+  private val processing = new AtomicBoolean(true)
+
+  private val unused = collection.mutable.Queue.fill(5)(createTopic())
 
-  val unused = collection.mutable.Queue.fill(5)(createTopic())
+  private def deleteTopics(): Unit = {
+    val toDelete = collection.mutable.Set.empty[String]
+    unneeded.drainTo(toDelete.asJava)
+    admin.deleteTopics(toDelete, 60*1000)
+  }
   
+  if(continuousCleanup) {
+    new Thread {
+      override def run {
+        while (processing.get) {
+          deleteTopics
+          Thread.sleep(60*1000)
```

Comment:
to use a schedular instead of sleep here is more thread efficient and CPU friendly, but as there is only one thread, should be ok